sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -158,25 +158,35 @@ static RecordEventsReadableSpan startSpan(
 
   @Override
   public SpanData toSpanData() {
+
+    // Copy immutable fields outside synchronized block.
     SpanContext spanContext = getSpanContext();
-    return SpanData.newBuilder()
-        .setName(getName())
-        .setInstrumentationLibraryInfo(instrumentationLibraryInfo)
-        .setTraceId(spanContext.getTraceId())
-        .setSpanId(spanContext.getSpanId())
-        .setTraceFlags(spanContext.getTraceFlags())
-        .setTracestate(spanContext.getTracestate())
-        .setAttributes(getAttributes())
-        .setStartEpochNanos(startEpochNanos)
-        .setEndEpochNanos(getEndEpochNanos())
-        .setKind(kind)
-        .setLinks(getLinks())
-        .setParentSpanId(parentSpanId)
-        .setHasRemoteParent(hasRemoteParent)
-        .setResource(resource)
-        .setStatus(getStatus())
-        .setTimedEvents(adaptTimedEvents())
-        .build();
+    SpanData.Builder builder =
+        SpanData.newBuilder()
+            .setName(getName())
+            .setInstrumentationLibraryInfo(instrumentationLibraryInfo)
+            .setTraceId(spanContext.getTraceId())
+            .setSpanId(spanContext.getSpanId())
+            .setTraceFlags(spanContext.getTraceFlags())
+            .setLinks(getLinks())
+            .setKind(kind)
+            .setTracestate(spanContext.getTracestate())
+            .setParentSpanId(parentSpanId)
+            .setHasRemoteParent(hasRemoteParent)
+            .setResource(resource)
+            .setStartEpochNanos(startEpochNanos);
+
+    // Copy remainder within synchronized
+    synchronized (lock) {
+      return builder
+          .setAttributes(getAttributes())
+          .setEndEpochNanos(getEndEpochNanos())
+          .setStatus(getStatus())
```

Comment:
getStatus is already synchronized, you probably want to call getStatusWithDefault here.